android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/BaseSearchTelemetry.kt

```diff
@@ -46,19 +77,25 @@ abstract class BaseSearchTelemetry {
             id = extensionInfo.id,
             url = extensionInfo.resourceUrl,
             onSuccess = { extension ->
-                store.flowScoped { flow -> subscribeToUpdates(flow, extension, extensionInfo) }
+                store.flowScoped { flow ->
+                    subscribeToUpdates(flow, extension, extensionInfo)
+                }
             },
             onError = { _, throwable ->
                 Logger.error("Could not install ${extensionInfo.id} extension", throwable)
             },
         )
+        CoroutineScope(Dispatchers.IO).launch {
+            waitUntilProviderListInitialized()
+        }
```

Comment:
I think we would benefit from separating the fetching of the providers from the installation of the engine, in terms of semantics.